packages/distributor-oracle/src/server.ts

```diff
@@ -437,10 +437,18 @@ export class OracleServer {
     }
 
     let _this = this;
-    const serializedTxs = await Promise.all(req.body.transactions.map(async (txData) => {
+    const serializedTxs = await Promise.all(req.body.transactions.map(async (txData, idx) => {
       const result = await _this.signTransaction(txData);
-      return result.success ? result.transaction : result.message;
+      if (!result.success) {
+        res
+          .status(400)
+          .send({ error: result.message ? `${result.message}\n\nTransaction index: ${idx}` : `Error signing transaction index: ${idx}` });
+        return null;
```

Comment:
I don't think this will do what you think it will. This return is inside the map, so the map just continues. Meanwhile you've already sent a 400. I'm not even sure what happens if you double res.send. Would prefer to not have that happen at all